src/core/Scene.ts

```diff
@@ -1204,6 +1208,27 @@ export class Scene extends ImperativeBase {
 			this.__elementParentSize = parentSize
 		}
 	}
+
+	#caster = new Raycaster()
+
+	#handleClick(ev: MouseEvent) {
+		const pointer = new Vector2()
+
+		pointer.x = (ev.clientX / this.clientWidth) * 2 - 1
+		pointer.y = -(ev.clientY / this.clientHeight) * 2 + 1
+
+		// Setup raycaster
+		this.#caster.setFromCamera(pointer, this.__threeCamera)
+		const intersections = this.#caster.intersectObject(this.three, true)
+		if (intersections.length == 0) return
+
+		// Create custom event with detail of the exact XYZ position of the click
+		const newEvent = new CustomEvent('click', {
+			detail: {position: intersections[0].point},
+		})
+		// Dispatch custom event on the LUME element saved on the intersection Three.js object
+		intersections[0].object.userData.lumeElement.dispatchEvent(newEvent)
```

Comment:
This will need to check up the three.js tree because there can be three.js subnodes and we need to find under which lume element those subnodes are

Reply:
Not super familiar with the three.js tree, but I assume that not every three.js `Object3d` will have a corresponding Lume element, so should it just check if `lumeElement` exists on `userData`, and if not, check the `Object3d`'s parent until one is found?

Reply:
Exactly, the element's `.three` property will be a Three.js node, and it might have Three.js children. This upward check could go in `getLumeElementFromThree` mentioned in the other comment.